include/nbl/builtin/glsl/ext/RadixSort/default_scatter.comp

```diff
@@ -0,0 +1,142 @@
+#define _NBL_GLSL_WORKGROUP_SIZE_ 256
+layout(local_size_x = _NBL_GLSL_WORKGROUP_SIZE_) in;
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_SET_DEFINED_
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_SET_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_BINDING_DEFINED_
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_BINDING_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_SET_DEFINED_
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_SET_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_BINDING_DEFINED_
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_BINDING_DEFINED_ 1
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_SET_DEFINED_
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_SET_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_BINDING_DEFINED_
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_BINDING_DEFINED_ 2
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_DESCRIPTOR_DEFINED_
+
+layout (set = _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_SET_DEFINED_, binding = _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_BINDING_DEFINED_, std430) readonly buffer in_buffer
+{
+	uvec2 in_values[];
+};
+
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_DESCRIPTOR_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_DESCRIPTOR_DEFINED_
+
+layout (set = _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_SET_DEFINED_, binding = _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_BINDING_DEFINED_, std430) writeonly buffer out_buffer
+{
+	uvec2 out_values[];
+};
+
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_DESCRIPTOR_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_DESCRIPTOR_DEFINED_
+
+layout (set = _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_SET_DEFINED_, binding = _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_BINDING_DEFINED_, std430) readonly buffer histogram_buffer
+{
+	uint histogram[];
+};
+
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_DESCRIPTOR_DEFINED_
+#endif
+
+#include "nbl/builtin/glsl/ext/RadixSort/parameters_struct.glsl"
+#include "nbl/builtin/glsl/ext/RadixSort/parameters.glsl"
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_PUSH_CONSTANTS_DEFINED_
+
+layout(push_constant) uniform PushConstants
+{
+	layout (offset = 0) nbl_glsl_ext_RadixSort_Parameters_t params;
+} pc;
+
+#define _NBL_GLSL_EXT_RADIXSORT_PUSH_CONSTANTS_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_GET_PARAMETERS_DEFINED_
+
+nbl_glsl_ext_RadixSort_Parameters_t nbl_glsl_ext_RadixSort_getParameters()
+{
+	return pc.params;
+}
+
+#define _NBL_GLSL_EXT_RADIXSORT_GET_PARAMETERS_DEFINED_
+#endif
+
+shared uint global_histogram[NUM_BUCKETS];
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_LOAD_GLOBAL_HISTOGRAM_DEFINED_
+
+void nbl_glsl_ext_RadixSort_loadGlobalHistogram()
+{
+	if (gl_LocalInvocationIndex < NUM_BUCKETS)
+		global_histogram[gl_LocalInvocationIndex] = histogram[gl_LocalInvocationIndex * gl_NumWorkGroups.x + gl_WorkGroupID.x];
+	barrier();
+}
```

Comment:
I wouldn't declare yet-another-shared-variable, I'd use `nbl_glsl_workgroupShuffle` but only have the first `NUM_BUCKETS` invocations perform useful loads as you do now.

Reply:
I'd do it like this
```glsl
uint nbl_glsl_ext_RadixSort_loadGlobalHistogram(in uint digit)
{
        uint offset;
	if (gl_LocalInvocationIndex<NUM_BUCKETS)
		offset = histogram[gl_LocalInvocationIndex * gl_NumWorkGroups.x + gl_WorkGroupID.x];
	return nbl_glsl_workgroupShuffle(offset,digit);
}
```